books/models.py

```diff
@@ -32,6 +32,47 @@ def cleanhtml(raw_html):
     return cleantext
 
 
+def get_book_data(book):
+    """Return the book data dict for a single Book instance, matching BookIndex.books property."""
+    has_faculty_resources = BookFacultyResources.objects.filter(book_faculty_resource=book).exists()
+    has_student_resources = BookStudentResources.objects.filter(book_student_resource=book).exists()
+    try:
+        return {
```

Comment:
this is what's returned on the current subjects page - let me know if you need other fields (or don't need them anymore 🔥)

Reply:
For the book tiles, the FE only needs id, cover_url, title, and slug. It pulls the rest of the details from the main book query (which is used for the subjects page).
Book tiles are currently generated from the book lists included in the specific-subjects page, so that's the model of list to use. The query for that is
```
`pages/${slug}-books?type=pages.Subject`
```

Reply:
@RoyEJohnson it doesn't make a separate query for every book tile does it?

Reply:
pretty sure it does, the console log was spitting out a constant stream of 
`Can't load Microbiology`
`Can't load A&P`
`...`

Reply:
Ok well id rather include extra data and have the option of changing it to like... Not do a separate query per tile if that's what it does

Reply:
It's not a separate query to the CMS for each book. The main subjects query is pulled once, and is [searched for the id](https://github.com/openstax/os-webview/blob/main/src/app/components/book-tile/book-tile.tsx#L58) of whatever book we're looking for.

Reply:
But will that work if we're searching for arbitrary books and we're not in the context of a particular subject? The API endpoint that you quoted before looked like it needed to be about a particular subject?

Reply:
here's a test page with the book block, it has nothing to do with subjects, it's just returning book metadata
https://dev.openstax.org/apps/cms/api/v2/pages/562/